plugins/UM3NetworkPrinting/src/Cloud/CloudApiClient.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+import json
+from json import JSONDecodeError
+from typing import Callable, List, Type, TypeVar, Union, Optional, Tuple, Dict
+
+from PyQt5.QtNetwork import QNetworkRequest, QNetworkReply
+
+from UM.Logger import Logger
+from cura.API import Account
+from cura.NetworkClient import NetworkClient
+from ..Models import BaseModel
+from .Models.CloudClusterResponse import CloudClusterResponse
+from .Models.CloudErrorObject import CloudErrorObject
+from .Models.CloudClusterStatus import CloudClusterStatus
+from .Models.CloudPrintJobUploadRequest import CloudPrintJobUploadRequest
+from .Models.CloudPrintResponse import CloudPrintResponse
+from .Models.CloudPrintJobResponse import CloudPrintJobResponse
+
+
+## The cloud API client is responsible for handling the requests and responses from the cloud.
+#  Each method should only handle models instead of exposing any HTTP details.
+class CloudApiClient(NetworkClient):
+
+    # The cloud URL to use for this remote cluster.
+    # TODO: Make sure that this URL goes to the live api before release
```

Comment:
We can use the new CuraConstants file that @LipuFei is adding in another PR (this issue was also there for the backups plugin that was moved to the Cura repository).